### PR TITLE
Push up to date

### PR DESCRIPTION
**Pull request by jgrund**
Add all commits up to master

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+            "last 1 chrome version",
+            "last 1 firefox version"
+          ]
+        }
+      }
+    ]
+  ],
+  "plugins": [
+    "transform-flow-strip-types",
+    "transform-strict-mode"
```
Comment (jgrund) [inline .babelrc]: don't need strict mode.

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,3 @@
 node_modules
-npm-debug.log
-.idea
-*.iml
+/coverage
+.nyc_output
```
Comment (jgrund) [inline .gitignore]: don't need .nyc_output.

File: .npmignore [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+.idea
+.eslintignore
+.editorconfig
+.eslintrc
+.flowconfig
+.babelrc
+.gitignore
+yarn.lock
+*.log
+run_test.sh
```
Comment (jgrund) [inline .npmignore]: can remove this.

File: .npmignore [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+readme.md
```
Comment (jgrund) [inline .npmignore]: remove this.

File: package.json [hunk 5]
```diff
@@ -5,28 +5,49 @@
...
+    "babel-plugin-transform-flow-strip-types": "^6.22.0",
+    "babel-preset-env": "^1.4.0",
+    "eslint": "^3.19.0",
+    "eslint-config-prettier": "^1.7.0",
+    "eslint-plugin-flowtype": "^2.32.1",
+    "eslint-plugin-import": "^2.2.0",
+    "eslint-plugin-prettier": "^2.0.1",
+    "flow-bin": "^0.44.2",
+    "jasmine": "2.6.0",
+    "jasmine-reporters": "^2.2.1",
```
Comment (jgrund) [inline package.json]: can remove this

File: package.json [hunk 6]
```diff
@@ -5,28 +5,49 @@
...
+    "babel-eslint": "^7.2.3",
+    "babel-plugin-transform-flow-strip-types": "^6.22.0",
+    "babel-preset-env": "^1.4.0",
+    "eslint": "^3.19.0",
+    "eslint-config-prettier": "^1.7.0",
+    "eslint-plugin-flowtype": "^2.32.1",
+    "eslint-plugin-import": "^2.2.0",
+    "eslint-plugin-prettier": "^2.0.1",
+    "flow-bin": "^0.44.2",
+    "jasmine": "2.6.0",
```
Comment (jgrund) [inline package.json]: can remove this

File: test/tests.test.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+// @flow
+
+import extractApi from '../index.js';
+import { format } from 'util';
+
+import { describe, it, expect } from './jasmine.js';
+
+describe('Extract API', function() {
+  ['/api/host/10', '/api/host/10/', 'api/host/10'].forEach(function(path) {
+    it(format('should extract the path id from %s', path), function() {
```
Comment (jgrund) [inline test/tests.test.js]: Can use es6 format strings.

File: .babelrc (hunk 1 above)
Comment (johnsonw) [inline .babelrc]: I'll remove. I originally saw it in fp so that's why I added it in.

File: .gitignore (hunk 2 above)
Comment (johnsonw) [inline .gitignore]: Ok, it's also in obj. Want me to remove it from there as well?

File: .npmignore (hunk 3 above)
Comment (johnsonw) [inline .npmignore]: removed

Comment (johnsonw) [inline .npmignore]: also in obj. Want me to remove from there as well?

File: .npmignore (hunk 4 above)
Comment (johnsonw) [inline .npmignore]: removed readme.md

File: package.json (hunk 6 above)
Comment (johnsonw) [inline package.json]: Removed jasmine. It's also in obj, want me to remove?

File: package.json (hunk 5 above)
Comment (johnsonw) [inline package.json]: removed jasmine-reporters

File: .gitignore (hunk 2 above)
Comment (jgrund) [inline .gitignore]: yes.

File: .npmignore (hunk 3 above)
Comment (jgrund) [inline .npmignore]: yes.

File: test/tests.test.js (hunk 7 above)
Comment (johnsonw) [inline test/tests.test.js]: updated

File: package.json (hunk 6 above)
Comment (jgrund) [inline package.json]: yes.

**Pull request merged**